darts/timeseries.py

```diff
@@ -3005,6 +3037,598 @@ def map(
+                        logger,
+                    )
+
+                transformation[
+                    "closed"
+                ] = "left"  # to garantee that the latest value is not included in the window: forecasting safe
+
+            comps_to_transform = (
+                set(transformation["components"])
+                if "components" in transformation
+                else original_components
+            )
+            comps_not_to_transform = original_components - comps_to_transform
+
+            if not self.is_deterministic:
+                filter_df_columns = [
+                    df_col
+                    for df_col in ts_df.columns.to_list()
+                    if re.sub(f"{sample_prefix}.*$", "", df_col) in comps_to_transform
+                ]
+
+                filter_df_non_transformed = [
+                    df_col
+                    for df_col in ts_df.columns.to_list()
+                    if re.sub(f"{sample_prefix}.*$", "", df_col)
+                    in comps_not_to_transform
+                ]
+            else:
+                filter_df_columns = [
+                    df_col
+                    for df_col in ts_df.columns.to_list()
+                    if df_col in comps_to_transform
+                ]
+                filter_df_non_transformed = [
+                    df_col
+                    for df_col in ts_df.columns.to_list()
+                    if df_col in comps_not_to_transform
+                ]
+
+            function_group_kwargs, function_kwargs = _get_kwargs(
+                transformation, _builtin_transforms
+            )
+
+            if isinstance(fn, str):  # built-in function
+                function_group = _builtin_transforms[fn][0].__name__
+                function_name = _builtin_transforms[fn][1]
+
+                new_columns.extend(
+                    [
+                        f"{fn}{transformation['window'] if 'window' in transformation else idx}_{comp_name}"
+                        for comp_name in comps_to_transform
+                    ]
+                )
+                # apply transformation to the selected components
+                resulting_transformations = pd.concat(
+                    [
+                        resulting_transformations,
+                        getattr(
+                            getattr(ts_df[filter_df_columns], function_group)(
+                                **function_group_kwargs
+                            ),
+                            function_name,
+                        )(**function_kwargs),
+                    ],
+                    axis=1,
+                )
+
+            elif isinstance(
+                transformation["function"], Callable
+            ):  # user provided function
+                new_columns.extend(
+                    [
+                        f"userFn{transformation['window'] if 'window' in transformation else idx}_{comp_name}"
```

Comment:
```suggestion
                        f"udf_{transformation['window'] if 'window' in transformation else idx}_{comp_name}"
```
("udf" is a common shorthand for "user-defined function")